### PR TITLE
Speeding up Rails boot time

### PR DESCRIPTION
**Pull request by hlmerscher**
This post goes through and explains the latest improvements regarding rails boot time on Columbus.

### CODE REVIEW

File: content/post/2019-11-20-speeding-up-rails-boot-time.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
++++
+title = "Speeding up Rails boot time"
+date = "2019-11-20"
+author = "Hercules Merscher"
+github = "hlmerscher"
++++
+
+Usually when we have a small Rails application, or are starting a new project, we don't have such a problem as slow boot time. At this point our app is using just a few libraries, logic has not evolved into a big monster, not many feature/integration tests... easy peasy!!!
+
+But then comes the day when you dread doing a change on your codebase. A single small change means an endless waiting time to boot the web app or to run a single unit test. I know, it is just "a few seconds", but it seems like an eternity when you need the good and old fast feedback of unit tests. Slow feedbacks during this process might end up frustrating team members trying to get things done.
```
Comment (lfritz) [inline content/post/2019-11-20-speeding-up-rails-boot-time.md]: `Slow feedbacks` should be `Slow feedback` (singular)

File: content/post/2019-11-20-speeding-up-rails-boot-time.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+Randomized with seed 46782
+
+real  1m 7.31s
+user  0m 6.38s
+sys   0m 10.01s
+```
+
+Not a pleasant experience to be honest. But we did some small tweaks and managed to improve this time drastically!
+
+It is not anything special or unknown to the Rails community. The things listed here comes by default on newer versions of Rails. But if you're still working on legacy versions of Rails, might be worth to take a look and see if you've done this already.
```
Comment (lfritz) [inline content/post/2019-11-20-speeding-up-rails-boot-time.md]: `The things listed here comes by default`: `come` should we without `s`

File: content/post/2019-11-20-speeding-up-rails-boot-time.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+gem 'bootsnap', require: false
+```
+
+And set it up on the `config/boot.rb` file:
+
+```ruby
+require 'bootsnap/setup'
+```
+
+Bootsnap does not clean up its own cache though. If you want to get around it and still use it in some situations you probably will want to add a toggle. One way would be such as:
```
Comment (lfritz) [inline content/post/2019-11-20-speeding-up-rails-boot-time.md]: `you probably will want to`: `you will probably want to` sounds better to me

File: content/post/2019-11-20-speeding-up-rails-boot-time.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+Add `require: false` for gems that don't need to be loaded automatically. Example:
+
+```ruby
+gem 'rubocop', require: false
+```
+
+Rubocop is on Gemfile just to support development and CI, to track the quality of code written. You don't need to load this gem on your Ruby app. We can save some memory and boot time by paying attention to this.
+
+Be mindful about dependencies. Libraries and frameworks are really helpful, and we need them to be productive at work, but there are some cases where we only need a function or a few lines of code, and in this case is ok to copy+paste from somewhere. As more dependencies we add, more constrained we are as time passes when dealing with upgrades, one library can depend of other(s) and if it stops being maintained we might end up having to dedicate more time to get rid of it later on.
```
Comment (lfritz) [inline content/post/2019-11-20-speeding-up-rails-boot-time.md]: > As more dependencies we add, more constrained we are...

I think either this:

> As we add more dependencies, we become more constrained...

or this:

> The more dependencies we add, the more constrained we are...

Comment (lfritz) [inline content/post/2019-11-20-speeding-up-rails-boot-time.md]: > there are some cases where we only need a function or a few lines of code, and in this case is ok to copy+paste from somewhere

Also, I completely agree with this :smile_cat:

File: content/post/2019-11-20-speeding-up-rails-boot-time.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+Any tips you'd like to share with us? Please let us know. :)
```
Comment (lfritz) [inline content/post/2019-11-20-speeding-up-rails-boot-time.md]: We don't have comments on the blog, actually... maybe we should add a link to Twitter?

Comment (hlmerscher) [reply content/post/2019-11-20-speeding-up-rails-boot-time.md]: Ok, I will remove the last sentence then :)

**Pull request merged**